File: Python/ImageLayer.py
class ImageLayer:
    def __init__(self, image):
        self.image = image
        self.dx = 0
        self.dy = 0
        self.sx = 1
        self.sy = 1
        self.r = 0

    def image(self):
        return self.image

    def clone(self):
        return ImageLayer(self.image.clone())

    def getPixel(self, i, j):
        return self.image.getPixel(i, j)

File: Java/ImageLayer.java
import java.awt.Color;

public class ImageLayer {
  Layer image;
  float dx = 0;
  float dy = 0;
  float sx = 1;
  float sy = 1;
  float r = 0;

  public ImageLayer(Layer image){
    this.image = image;
  }

  public Layer image(){
    return this.image;
  }

  public ImageLayer clone(){
    ImageLayer toReturn = new ImageLayer(this.image.clone());
    return toReturn;
  }

  public Color getPixel(int i, int j) {
    return image.getPixel(i,j);
  }

  

}
